### PR TITLE
bug 1875038: endpointaccessible: do not panic when no endpoints are found

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

File: pkg/controllers/endpointaccessible/endpoint_accessible_controller.go [hunk 1]
```diff
@@ -55,6 +55,12 @@ func (c *endpointAccessibleController) sync(ctx context.Context, syncCtx factory
 		return err
 	}
 
+	// the endpoint list function can return empty results without error when the endpoints does not exists
+	// requeue in that case and wait until we get valid endpoints to check.
+	if len(endpoints) == 0 {
+		return factory.SyntheticRequeueError
```
Comment (deads2k) [inline pkg/controllers/endpointaccessible/endpoint_accessible_controller.go:61]: if this is the case, we should be degraded.  This shouldn't be a synthentic error.

Comment (deads2k) [reply pkg/controllers/endpointaccessible/endpoint_accessible_controller.go:61]: > if this is the case, we should be degraded. This shouldn't be a synthentic error.

actuall, set the AVailable to false.

Comment (mfojtik) [reply pkg/controllers/endpointaccessible/endpoint_accessible_controller.go]: @deads2k updated, we go available=false in case no endpoints are returned.

**Pull request merged**